### PR TITLE
harmonize URL/URI

### LINKED ISSUE #341: Use of the terms URI and URL
In various places in the protocol, the terms _URI_ and _URL_ seem to me to be used more or less synonymously. For example, in [Section 3.1](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-3.1) there is a property `uri` in the `continue` field, while in [Section 3.3.3](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-3.3.3) there is a member `url` in the `user_code` field. Another example is the description of the `redirect` interaction start mode. While in Sections [2.5.1.1](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-2.5.1.1) and [3.3.1](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-3.3.1) the term _URL_ is used consistently, in [Section 4.1.1](https://datatracker.ietf.org/doc/html/draft-ietf-gnap-core-protocol#section-4.1.1) mostly _URI_ is used, but in one sentence _URL_ is used again.

Wouldn't it make sense to make this consistent and always speak of a _URL_, since I assume that the use of Uniform Resource Names (_URNs_) is not intended? Or is there a reason why _URI_ is used in some places and _URL_ in others?

### PR DESCRIPTION
**Pull request by fimbault**
The aim is to Close #341

### CODE REVIEW

File: draft-ietf-gnap-core-protocol.md [hunk 1]
```diff
@@ -1818,12 +1818,19 @@ NOTE: '\' line wrapping per RFC 8792
 }
 ~~~
 
-In this example, the AS is returning a bearer [access token](#response-token-single) with a management URL and a [subject identifier](#response-subject) in the form of
+In this example, the AS is returning a bearer [access token](#response-token-single) with a management URI and a [subject identifier](#response-subject) in the form of
+an opaque identifier.
+
+~~~
+NOTE: '\' line wrapping per RFC 8792 and a [subject identifier](#response-subject) in the form of
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:1825]:
```suggestion
In this example, the AS is returning a bearer [access token](#response-token-single) with a management URI and a [subject identifier](#response-subject) in the form of
```

File: draft-ietf-gnap-core-protocol.md [hunk 2]
```diff
@@ -1818,12 +1818,19 @@ NOTE: '\' line wrapping per RFC 8792
...
 an opaque identifier.
 
 ~~~
 NOTE: '\' line wrapping per RFC 8792
 
+{
+    "access_token": {
+
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:1833]:
```suggestion
```

File: draft-ietf-gnap-core-protocol.md [hunk 3]
```diff
@@ -2243,17 +2250,17 @@ code (string)
               minutes. It is RECOMMENDED that this code be no more than eight
               characters in length.
 
-url (string)
-: RECOMMENDED. The interaction URL that the client instance
-              will direct the RO to. This URL MUST be stable such
+uri (string)
+: RECOMMENDED. The interaction URI that the client instance
+              will direct the RO to. This URI MUST be stable such
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:2255]: Revert parameter name change, will be addressed in with #123

File: draft-ietf-gnap-core-protocol.md [hunk 4]
```diff
@@ -2243,17 +2250,17 @@ code (string)
...
               that client instances can be statically configured with it.
 
 
 ~~~
 "interact": {
     "user_code": {
         "code": "A1BC-3DFF",
-        "url": "https://srv.ex/device"
+        "uri": "https://srv.ex/device"
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:2263]:
```suggestion
        "url": "https://srv.ex/device"
```

File: draft-ietf-gnap-core-protocol.md [hunk 5]
```diff
@@ -4521,10 +4528,10 @@ containing the following information:
 
 grant_request_endpoint (string)
 : REQUIRED. The location of the
-          AS's grant request endpoint. The location MUST be a URL {{RFC3986}}
+          AS's grant request endpoint. The location MUST be a URI {{RFC3986}}
           with a scheme component that MUST be https, a host component, and optionally,
-          port, path and query components and no fragment components. This URL MUST
-          match the URL the client instance used to make the discovery request.
+          port, path and query components and no fragment components. This URI MUST
+          match the URI the client instance used to make the discovery request.
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:4534]: Revert, this is a URL

File: draft-ietf-gnap-core-protocol.md [hunk 6]
```diff
@@ -5437,25 +5444,36 @@ otherwise unexposed endpoints.
...
+a URI that is within reach of the target system but normally unavailable to the attacker. For
+example, an attacker sending a URL of `http://localhost/admin` to cause the server to access an
+internal function on itself, or `https://192.168.0.14/` to call a service behind a firewall.
+Even if the attacker does not gain access to the results of the call, the side effects of such
+requests coming from a trusted host can be problematic to the security and sanctity of such
+otherwise unexposed endpoints.
+
+In GNAP, the most vulnerable place in the core protocol is the
+[push-based post-interaction finish method](#interaction-pushback), as the client instance is
+less trusted than the AS and can use this method to make the AS calys from a presented URI instead of the client instance presenting the key by value.
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:5463]: Review this text, does not seem to be an intentional edit.

File: draft-ietf-gnap-core-protocol.md [hunk 7]
```diff
@@ -5774,7 +5792,7 @@ Cache-Control: no-store
 
 
 The client instance saves the response and redirects the user to the
-interaction_url by sending the following HTTP message to the user's
+interaction_uri by sending the following HTTP message to the user's
```
Comment (jricher) [inline draft-ietf-gnap-core-protocol.md:5795]:
```suggestion
redirect interaction start URI by sending the following HTTP message to the user's
```

File: draft-ietf-gnap-core-protocol.md (hunk 3 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md:2255]: Reverted

File: draft-ietf-gnap-core-protocol.md (hunk 7 above)
Comment (fimbault) [reply draft-ietf-gnap-core-protocol.md]: Reverted, not sure what the suggested change was about

**Pull request merged**